lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -1449,4 +1516,117 @@ class TTIRLoadSystemDesc
+    auto ty = mlir::cast<RankedTensorType>(input.getType());
+    auto output =
+        rewriter.create<tensor::EmptyOp>(loc, shapei64, ty.getElementType());
+
+    auto shape_attr = rewriter.getI32ArrayAttr(
+        {static_cast<int32_t>(shapei64[0]), static_cast<int32_t>(shapei64[1]),
+         static_cast<int32_t>(shapei64[2]), static_cast<int32_t>(shapei64[3])});
+    return rewriter.create<ttir::ReshapeOp>(
+        loc, output.getType(), input, output, shape_attr, operandConstraints);
+  }
+
+  LogicalResult matchAndRewrite(MeanOp reduce1,
+                                PatternRewriter &rewriter) const final {
+
+    if (!reduce1.getDimArg().has_value()) {
+      return failure();
+    }
+
+    if (reduce1.getDimArg().value().size() != 1) {
+      return failure();
+    }
+
+    int64_t reduce1_dim =
+        mlir::cast<IntegerAttr>(reduce1.getDimArg().value().getValue()[0])
+            .getInt();
+    if (reduce1_dim != -3) {
+      return failure();
+    }
+
+    MeanOp reduce2;
+    bool found = false;
+    for (auto user : reduce1->getUsers()) {
+      user->getName().dump();
+      if (isa<MeanOp>(user)) {
+        reduce2 = mlir::cast<MeanOp>(user);
+        found = true;
+      }
+    }
+    if (!found) {
+      return failure();
+    }
+
+    if (!reduce2.getDimArg().has_value()) {
+      return failure();
+    }
+
+    if (reduce2.getDimArg().value().size() != 1) {
+      return failure();
+    }
+
+    int64_t reduce2_dim =
+        mlir::cast<IntegerAttr>(reduce2.getDimArg().value().getValue()[0])
+            .getInt();
+    if (reduce2_dim != -2) {
+      return failure();
+    }
+
+    auto input = reduce1->getOperand(0);
+    auto input_type = mlir::cast<RankedTensorType>(input.getType());
+    std::vector<int64_t> input_shape = input_type.getShape();
+    input_shape[input_shape.size() - 2] = input_shape[input_shape.size() - 2] *
+                                          input_shape[input_shape.size() - 3];
+    input_shape[input_shape.size() - 3] = 1;
+
+    llvm::ArrayRef<int64_t> shape_attr = {
+        (int32_t)input_shape[0], (int32_t)input_shape[1],
+        (int32_t)input_shape[2], (int32_t)input_shape[3]};
+
+    // auto to_layout_dps =
+    //     rewriter.create<tensor::EmptyOp>(reduce1->getLoc(),
+    //     input_type.getShape(), input_type.getElementType(), );
+
+    // auto layout_attr = mlir::cast<LayoutAttr>(input_type.getEncoding());
+    // layout_attr = layout_attr,
+    // auto to_layout = rewriter.create<ttir::ToLayoutOp>(reduce1.getLoc(),
+    // to_layout_dps.getType(), input, LayoutAttr::get(rewriter.getContext(),
+    // ));
+
+    auto reshape = createReshapeOp(rewriter, reduce1->getLoc(), input,
+                                   shape_attr, reduce1.getOperandConstraints());
+
+    rewriter.replaceOp(reduce1, reshape);
+    rewriter.modifyOpInPlace(reduce2,
+                             [&]() { reduce2.setOperand(0, reshape); });
```

Comment:
Is this `modifyOpInPlace` call really needed? The `replaceOp` call above should automatically do this.